projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-flag/components/case-flag-summary-list/case-flag-summary-list.component.ts

```diff
@@ -43,4 +46,33 @@ export class CaseFlagSummaryListComponent implements OnInit {
     const subTypeValue = flagDetail.subTypeValue ? ` - ${flagDetail.subTypeValue}` : '';
     return `${flagDetail.name}${otherDescription}${subTypeValue}`;
   }
+
+  private getHeaderText(): string {
+    if (this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE ||
+      this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE_EXTERNAL) {
+      return this.addFlagHeaderText;
+    }
+    if (this.displayContextParameter === CaseFlagDisplayContextParameter.UPDATE ||
+      this.displayContextParameter === CaseFlagDisplayContextParameter.UPDATE_EXTERNAL) {
+      return this.updateFlagHeaderText;
+    }
+    return '';
+  }
+
+  private getSummaryListDisplayMode(): number {
+    if (this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE ||
+        this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE_EXTERNAL) {
+      return CaseFlagSummaryListDisplayMode.CREATE;
+    }
+    return CaseFlagSummaryListDisplayMode.MANAGE;
+  }
+
+  private getCanDisplayStatus(): boolean {
+    if (this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE_EXTERNAL ||
```

Comment:
@johnbenjamin-hmcts - I think this can be simplified to `return !(this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE_EXTERNAL || this.displayContextParameter === CaseFlagDisplayContextParameter.UPDATE_EXTERNAL || this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE);`

Reply:
@Daniel-Lam - I have updated the code as you suggested.